src/app/shared/services/media-state.service.ts

```diff
@@ -0,0 +1,31 @@
+import { lastValueFrom } from 'rxjs'
+import { Injectable } from '@angular/core'
+import { MediaService } from './media.service'
+import { Media } from './media'
+import { Logger } from '../../core/abstractions/logger.service'
+
+@Injectable()
+export class MediaStateService {
+  private readonly logger: Logger
+
+  constructor(
+    private readonly mediaService: MediaService,
+    logger: Logger
+  ) {
+    this.logger = logger.tag('MediaStateService')
+  }
+
+  public async getMedia(id: string): Promise<Media | undefined> {
+    let media: Media | undefined = undefined
+    try {
+      media = await this.fetchMedia(id)
+    } catch (error) {
+      this.logger.error(`Failed to fetch media with id: ${id}, reason: ${error}`)
+    }
+    return media
+  }
```

Comment:
The HttpMediaService is already catching the error if we fail to find one, you don't need to do it again here
![image](https://github.com/tv2/ng-sofie/assets/23049647/b97df921-7f0e-4384-8169-38d10c81ca42)

But I wouldn't spend much time on the `MediaStateService` for now. SOF-1652 is going to set it up with listening for Media events etc and I believe @RasmusAlbrektsen is going to pick up that task